src/main/java/biz/paluch/logging/gelf/log4j2/GelfDynamicMdcFieldType.java

```diff
@@ -0,0 +1,59 @@
+package biz.paluch.logging.gelf.log4j2;
+
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.core.config.Configuration;
+import org.apache.logging.log4j.core.config.plugins.Plugin;
+import org.apache.logging.log4j.core.config.plugins.PluginAttribute;
+import org.apache.logging.log4j.core.config.plugins.PluginConfiguration;
+import org.apache.logging.log4j.core.config.plugins.PluginFactory;
+import org.apache.logging.log4j.status.StatusLogger;
+import org.apache.logging.log4j.util.Strings;
+
+import java.util.regex.Pattern;
+
+/**
+ * Configuration for dynamic log fields pulled from MDC.
+ *
+ * @author Thomas Herzog
+ */
+@Plugin(name = "DynamicMdcFieldTypes", category = "Core", printObject = true)
+public class GelfDynamicMdcFieldType {
+
+    private static final Logger LOGGER = StatusLogger.getLogger();
+
+    private final String regex;
+    private final String type;
+
+    public GelfDynamicMdcFieldType(String regex, String type) {
+        this.regex = regex;
+        this.type = type;
+    }
+
+    public String getRegex() {
+        return regex;
+    }
+
+    public String getType() {
+        return type;
+    }
+
+    @PluginFactory
+    public static GelfDynamicMdcFieldType createField(@PluginConfiguration final Configuration config,
+                                                      @PluginAttribute("regex") String regex,
+                                                      @PluginAttribute("type") String type) {
+
+        if (Strings.isEmpty(regex) || Strings.isEmpty(type)) {
```

Comment:
Let's throw exceptions instead of returning `null` values to signal a failure. Having additional log statements is helpful.